### DISCUSSION TITLE
What's the recommended way to run multiple JSON.SET commands?

### DISCUSSION THREAD

**Discussion by mauriciocirelli** [NodeRedis/node_redis#2741] [Q&A]
Dear  community,

I would like to ask what's the recommended way to run multiple `JSON.SET` commands.
Do we have a `multi()` operator for it? I could not find any...

I have seen an old thread doing a `Promise.all([])` call to an array of `client.json.set()` calls. Is this the way to go? I think this way it would not run the commands in a transaction, right?

Thank you!

**Comment by leibale** [comment]
1. Use the [`JSON.MSET` command](https://redis.io/docs/latest/commands/json.mset/) instead of `JSON.SET`:
```javascript
await client.json.mSet([{
  key: '1',
  path: '$',
  value: '1'
}, {
  key: '2',
  path: '$',
  value: '2'
}]);
```
2. Use `Promise.all`. but as you've said, it won't run the commands in a "transaction":
```javascript
const entries = [{
  key: '1',
  path: '$',
  value: '1'
}, {
  key: '2',
  path: '$',
  value: '2'
}];

await Promise.all(
  entries.map(({ key, path, value }) => client.json.set(key, path, value)
);
```
3. Use `client.multi()`:
```javascript
const entries = [{
  key: '1',
  path: '$',
  value: '1'
}, {
  key: '2',
  path: '$',
  value: '2'
}];

const multi = client.multi();

for (const { key, path, value } of entries) {
  multi.json.set(key, path, value);
}

const replies = await multi.exec();
```

**Comment by mauriciocirelli** [comment]
Excellent!, `multi.json.set()` is what I was looking for...
Thank you!
